docker/Dockerfile.memcheck

```diff
@@ -0,0 +1,30 @@
+FROM fedora:37
+
+RUN dnf -y update \
+    && dnf -y install \
+        gcc-fortran \
+        gcc-c++ \
+        gcc \
+        gdb \
+        git \
+        netcdf-fortran-devel \
+        cmake \
+        make \
+        lcov \
+        valgrind \
+        python3 \
+        python3-pip \
+        lapack-devel \
+        yaml-cpp-devel \
+    && dnf clean all
+
+RUN pip3 install numpy scipy
+
+# build the tuv-x tool
+COPY . /tuv-x/
+RUN mkdir /build \
+      && cd /build \
+      && cmake /tuv-x \
+      && make -j 8
```

Comment:
Do we now need the `ENABLE_MEMCHECK:BOOL=TRUE` here?

Reply:
and, actually you might need to merge main into this branch, because I think some changes came in with #51 that affect some of the docker files, and may not auto-merge well (like removing the parallel testing that doesn't work with the musica-core code)